Content.Server/Sticky/Systems/StickySystem.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Content.Server.DoAfter;
+using Content.Server.Popups;
+using Content.Server.Sticky.Components;
+using Content.Server.Sticky.Events;
+using Content.Shared.Interaction;
+using Robust.Server.GameObjects;
+using Robust.Shared.Containers;
+using Robust.Shared.Player;
+
+namespace Content.Server.Sticky.Systems;
+
+public sealed class StickySystem : EntitySystem
+{
+    [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly SharedContainerSystem _containerSystem = default!;
+
+    private const string StickerSlotId = "stickers_container";
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<StickySuccessfulEvent>(OnStickSuccessful);
+        SubscribeLocalEvent<StickyComponent, AfterInteractEvent>(OnAfterInteract);
+    }
+
+    private void OnAfterInteract(EntityUid uid, StickyComponent component, AfterInteractEvent args)
+    {
+        if (args.Handled || args.Target == null)
+            return;
+
+        // show message to user
+        if (component.StickPopupStart != null)
+        {
+            var msg = Loc.GetString(component.StickPopupStart);
+            _popupSystem.PopupEntity(msg, args.User, Filter.Entities(args.User));
+        }
+
+        // start sticking object to target
+        var delay = (float) component.Delay.TotalSeconds;
+        _doAfterSystem.DoAfter(new DoAfterEventArgs(args.User, delay, target: args.Target)
+        {
+            BroadcastFinishedEvent = new StickySuccessfulEvent(args.User, args.Target.Value, component),
```

Comment:
imo dont pass the component into the event, since it can get deleted etc during the doafter. just let it get resolved in `StickToEntity`